cmd/utils/cmd.go

```diff
@@ -85,6 +89,24 @@ func StartNode(stack *node.Node) {
 	}()
 }
 
+func monitorFreeDiskSpace(sigc chan os.Signal, path string) {
+	for {
+		freeSpace, err := getFreeDiskSpace(path)
+		if err != nil {
+			log.Warn("Failed to get free disk space", "path", path, "err", err)
+			break
+		}
+		if freeSpace < freeDiskSpaceCritical {
+			log.Error("Low disk space. Gracefully shutting down Geth to prevent database corruption.", "available", common.StorageSize(freeSpace))
+			sigc <- syscall.SIGTERM
+			break
+		} else if freeSpace < freeDiskSpaceWarning {
+			log.Warn("Disk space is running low. Geth will shutdown if disk space runs below critical level.", "available", common.StorageSize(freeSpace), "critical_level", common.StorageSize(freeDiskSpaceCritical))
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
Every 5 seconds is a bit too often, IMO. The warnings will hog the output. Maybe once a minute? That's roughly every fourth block

Reply:
Sure. Maybe Geth should auto-shutdown/quit if disk space is low though?

Reply:
> Sure. Maybe Geth should auto-shutdown/quit if disk space is low though? (maybe with log.Fatal) ?

.... That's basically the purpose of this PR

Reply:
Sorry

Reply:
(and log.Fatal is not a good mechanism, hence why this PR uses the same channel as ctrl-c uses. log.Fatal abruptly quits, without giving geth a chance to save stuff and close the db correctly)

Reply:
fixed